src/malco/post_process/mondo_score_utils.py

```diff
@@ -35,56 +26,56 @@ def omim_mappings(term: str) -> List[str]:
 
     Returns:
         str: The OMIM mappings.
-    """
-    adapter = mondo_adapter()
-    if not isinstance(adapter, MappingProviderInterface):
-        raise ValueError("Adapter is not an MappingProviderInterface")
+    """   
     omims = []
-    for m in adapter.sssom_mappings([term], "OMIM"):
+    for m in adapter.sssom_mappings([term], source="OMIM"):
         if m.predicate_id == "skos:exactMatch":
             omims.append(m.object_id)
     return omims
 
 
-def score_grounded_result(prediction: str, ground_truth: str) -> float:
+@cached(cache=LRUCache(maxsize=4096), info=True, key=lambda prediction, ground_truth, mondo: hashkey(prediction, ground_truth))
+def score_grounded_result(prediction: str, ground_truth: str, mondo) -> float:
     """
     Score the grounded result.
 
     Exact match:
-
     >>> score_grounded_result("OMIM:132800", "OMIM:132800")
     1.0
 
     The predicted Mondo is equivalent to the ground truth OMIM
     (via skos:exactMatches in Mondo):
-
     >>> score_grounded_result("MONDO:0007566", "OMIM:132800")
     1.0
 
     The predicted Mondo is a disease entity that groups multiple
     OMIMs, one of which is the ground truth:
-
     >>> score_grounded_result("MONDO:0008029", "OMIM:158810")
     0.5
 
     Args:
         prediction (str): The prediction.
         ground_truth (str): The ground truth.
+        mondo: the mondo adapter.
 
     Returns:
         float: The score.
     """
+    if not isinstance(mondo, MappingProviderInterface):
+        raise ValueError("Adapter is not an MappingProviderInterface")
+    
     if prediction == ground_truth:
         # predication is the correct OMIM
         return FULL_SCORE
-    if ground_truth in omim_mappings(prediction):
+
+    #if ground_truth in omim_mappings(prediction, mondo):
+    if ground_truth in omim_mappings(prediction, mondo):
         # prediction is a MONDO that directly maps to a correct OMIM
         return FULL_SCORE
-    mondo = mondo_adapter()
 
     descendants_list = mondo.descendants([prediction], predicates=[IS_A], reflexive=True)
     for mondo_descendant in descendants_list:
-        if ground_truth in omim_mappings(mondo_descendant):
+        if ground_truth in omim_mappings(mondo_descendant, mondo):
             # prediction is a MONDO that maps to a correct OMIM via a descendant
             return PARTIAL_SCORE
```

Comment:
It seems that we should provide full score (full rank) if a disease is in the same phenotypic series, but not necessarily if it is a Mondo descendent. I am not sure if that is what the code above is doing, but if so it would be good to add a comment and an example of what does and does not count (lines 72-74). I am not sure that we want to return a partial score for anything else, and it would really be better just to report ranks